ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -357,39 +417,79 @@ func getReplicaIndex(clusterName string, groupName string, namespace string) int
 }
 
 // returns next lowest TPU_WORKER_ID in pod slice and updates mappings
-func getNextWorkerID(podSlice slice, namespace string, replicaIndex int) int {
-	tpuWorkerID := 0
-	if sliceToWorkers[podSlice] == nil {
-		newWorker := worker{tpuWorkerID, replicaIndex, true}
-		sliceToWorkers[podSlice] = []worker{newWorker}
-	} else {
-		nextLowestID := math.MaxInt32
-		replacePod := false
-		// iterate through existing workers and check if any have been deleted
-		for _, worker := range sliceToWorkers[podSlice] {
-			if worker.isCreated == false && worker.workerIndex < nextLowestID {
-				replacePod = true
-				nextLowestID = worker.workerIndex
+func getNextWorkerID(sliceToWorkerIDs map[slice][]int, podSlice slice, namespace string, replicaIndex int) int {
+	tpuWorkerID := 0 // defaults to 0 (first Pod in slice)
+	if sliceToWorkerIDs[podSlice] != nil {
```

Comment:
In general, try to avoid multiple indentations if possible. I find it's easier to follow code with less indentations. This function could be updated to the following:
```go
// returns next lowest TPU_WORKER_ID in pod slice and updates mappings
func getNextWorkerID(sliceToWorkerIDs map[slice][]int, podSlice slice, namespace string, replicaIndex int) int {
	tpuWorkerID := 0 // defaults to 0 (first Pod in slice)
	if sliceToWorkerIDs[podSlice] == nil {
	    return tpuWorkerID
	}
	
        sort.Ints(sliceToWorkerIDs[podSlice])
        // iterate through existing workers and get the next lowest, unused ID
        for _, workerID := range sliceToWorkerIDs[podSlice] {
	        if workerID == tpuWorkerID {
		        tpuWorkerID++
                         // break here?
	        }
        }

	klog.V(1).InfoS("getNextWorkerID", "RayCluster", namespace+"/"+podSlice.clusterName, "Worker Group", podSlice.groupName, "TPU_WORKER_ID", tpuWorkerID)
	return tpuWorkerID
}
```

Reply:
Sounds good, I'll follow that style from now on. I added this change in [221d595](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/221d5950ee8b8387aa43bf63fd115a3ad45a0ad6).

Reply:
We break early now in df2ed78.